src/Shared/BuildEnvironmentHelper.cs

```diff
@@ -550,9 +550,16 @@ public BuildEnvironment(BuildEnvironmentMode mode, string currentMSBuildExePath,
             {
                 // In the .NET SDK, there's one copy of MSBuild.dll and it's in the root folder.
                 MSBuildToolsDirectoryRoot = CurrentMSBuildToolsDirectory;
+
+                // If we're standalone, we might not be in the SDK. Rely on folder paths at this point.
+                if (string.Equals(currentToolsDirectory.Name, "amd64", StringComparison.OrdinalIgnoreCase) ||
+                    string.Equals(currentToolsDirectory.Name, "arm64", StringComparison.OrdinalIgnoreCase))
+                {
+                    MSBuildToolsDirectoryRoot = currentToolsDirectory.Parent?.FullName;
+                }
             }
 
-            if (mode == BuildEnvironmentMode.VisualStudio && MSBuildToolsDirectoryRoot != null)
+            if (MSBuildToolsDirectoryRoot != null)
```

Comment:
Can you explain taking out the VS check here?

Reply:
We mistakenly put this in because we only considered two scenarios:
1. VS
2. SDK

In the sdk scenario, there is no arm64/ or amd64/ folder. The VS scenario however needs to dig a little bit to find the others.

What we didn't realize is that the corext scenario also needed this logic.

Reply:
So should this say mode != BuildEnvironmentMode.SDK then?

Reply:
If there was one, yep! This change should be minimal though, since we're servicing